### PR TITLE
e2e: dump conditions, and velero pod log errors

### PR DESCRIPTION
**Pull request by kaovilai**
Use GInkgoWriter to append logs to display if a test fails.
Output conditions and velero pod errors to ginkgo error output.

Get helpful logs like directly in ginkgo output.
```
2021/10/10 02:14:03 Pod not running with condition: containers with unready status: [mssql-app]
2021/10/10 02:14:08 Pod not running with condition: 0/6 nodes are available: 1 node(s) had volume node affinity conflict, 2 Insufficient cpu, 3 node(s) had taint {node-role.kubernetes.io/master: }, that the pod didn't tolerate.
```
```
• Failure [50.187 seconds]
AWS backup restore tests
/Users/tiger/git/oadp-operator/tests/e2e/backup_restore_suite_test.go:16
  backup and restore applications
  /Users/tiger/git/oadp-operator/tests/e2e/backup_restore_suite_test.go:53
    MSSQL application [It]
    /Users/tiger/git/oadp-operator/tests/e2e/backup_restore_suite_test.go:123

    Expected
        <[]string | len:28, cap:32>: [
            "velero container error line#80: time=\"2021-10-11T01:40:54Z\" level=error msg=\"Error listing backups in backup store\" backupLocation=ts-example-velero-1 controller=backup-sync error=\"rpc error: code = Unknown desc = NoSuchBucket: The specified bucket does not exist\\n\\tstatus code: 404, request id: QJST2JGX6WE8SK3E, host id: sNuMNaTnZ1pq83kUdDNw5nCc/w6sE/tzmu0k25WxLo7585nx9W2hNt8phGJGxWz0Q4Hy+QjCQBM=\" error.file=\"/go/src/velero-plugin-for-aws/velero-plugin-for-aws/object_store.go:361\" error.function=\"main.(*ObjectStore).ListCommonPrefixes\" logSource=\"pkg/controller/backup_sync_controller.go:182\"\n",
            "velero container error line#81: time=\"2021-10-11T01:40:54Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0, Backup storage location \\\"ts-example-velero-1\\\" is unavailable: rpc error: code = Unknown desc = NoSuchBucket: The specified bucket does not exist\\n\\tstatus code: 404, request id: QJSR9EFMSEFGZ45E, host id: bzcfbvHz0VmvWKr9pvu65+KBFFvkR+6lUsg6T8zWjPcC0CXUGy8MgAe21EaZsObVIrKtM6CdGUQ=)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:164\"\n",
            "velero container error line#82: time=\"2021-10-11T01:40:54Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#83: time=\"2021-10-11T01:40:54Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#84: time=\"2021-10-11T01:40:54Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#85: time=\"2021-10-11T01:40:54Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#86: time=\"2021-10-11T01:40:54Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#87: time=\"2021-10-11T01:40:54Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#88: time=\"2021-10-11T01:40:55Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#90: time=\"2021-10-11T01:40:55Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#91: time=\"2021-10-11T01:40:57Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#92: time=\"2021-10-11T01:40:59Z\" level=error msg=\"Current backup storage locations available/unavailable/unknown: 0/1/0)\" controller=backup-storage-location logSource=\"pkg/controller/backup_storage_location_controller.go:166\"\n",
            "velero container error line#104: time=\"2021-10-11T01:41:04Z\" level=error msg=...
    
    Gomega truncated this representation as it exceeds 'format.MaxLength'.
    Consider having the object provide a custom 'GomegaStringer' representation
    or adjust the parameters in Gomega's 'format' package.
    
    Learn more here: https://onsi.github.io/gomega/#adjusting-output
    
    to equal
        <[]string | len:0, cap:0>: []

    /Users/tiger/git/oadp-operator/tests/e2e/backup_restore_suite_test.go:82
```

### CODE REVIEW

**Comment by kaovilai** [comment]
Very weird error in restore results in objectStore from https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_oadp-operator/381/pull-ci-openshift-oadp-operator-master-operator-e2e/1447403911777357824
```json
{
    "errors": {
        "cluster": [
            "error preparing securitycontextconstraints.security.openshift.io/mssql-persistent-scc: rpc error: code = Unknown desc = BackupStorageLocation not found"
        ],
        "namespaces": {
            "mssql-persistent": [
                "error preparing persistentvolumeclaims/mssql-persistent/mssql-pvc: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/builder-dockercfg-d96h5: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/builder-token-bcxbm: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/builder-token-ktk7w: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/default-dockercfg-4f6bl: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/default-token-8g8mm: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/default-token-bfwbv: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/deployer-dockercfg-4qsl6: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/deployer-token-4bqvr: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/deployer-token-bmdkn: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/mssql-persistent-sa-dockercfg-hg9hg: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/mssql-persistent-sa-token-ld88w: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/mssql-persistent-sa-token-vpf8f: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing secrets/mssql-persistent/mssql-secret: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing configmaps/mssql-persistent/kube-root-ca.crt: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing configmaps/mssql-persistent/openshift-service-ca.crt: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing serviceaccounts/mssql-persistent/builder: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing serviceaccounts/mssql-persistent/default: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing serviceaccounts/mssql-persistent/deployer: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing serviceaccounts/mssql-persistent/mssql-persistent-sa: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing pods/mssql-persistent/mssql-app-deployment-6dbc8d5b64-hwgjt: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing pods/mssql-persistent/mssql-deployment-1-mf954: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing replicasets.apps/mssql-persistent/mssql-app-deployment-6dbc8d5b64: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing deploymentconfigs.apps.openshift.io/mssql-persistent/mssql-deployment: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing deployments.apps/mssql-persistent/mssql-app-deployment: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing endpoints/mssql-persistent/mssql-app-service: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing endpoints/mssql-persistent/mssql-service: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing endpointslices.discovery.k8s.io/mssql-persistent/mssql-app-service-v4pc8: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing endpointslices.discovery.k8s.io/mssql-persistent/mssql-service-xbgtt: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing replicationcontrollers/mssql-persistent/mssql-deployment-1: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing rolebindings.authorization.openshift.io/mssql-persistent/system:deployers: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing rolebindings.authorization.openshift.io/mssql-persistent/system:image-builders: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing rolebindings.authorization.openshift.io/mssql-persistent/system:image-pullers: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing rolebindings.rbac.authorization.k8s.io/mssql-persistent/system:deployers: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing rolebindings.rbac.authorization.k8s.io/mssql-persistent/system:image-builders: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing rolebindings.rbac.authorization.k8s.io/mssql-persistent/system:image-pullers: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing routes.route.openshift.io/mssql-persistent/mssql-app-route: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing services/mssql-persistent/mssql-app-service: rpc error: code = Unknown desc = BackupStorageLocation not found",
                "error preparing services/mssql-persistent/mssql-service: rpc error: code = Unknown desc = BackupStorageLocation not found"
            ]
        }
    },
    "warnings": {}
}
```

Possibly came from https://github.com/openshift/openshift-velero-plugin/blob/dc9eefb109cf187567ab65423f40d7a6faa65abc/velero-plugins/common/shared.go#L192

**Comment by kaovilai** [comment]
From reviewing repeating logs.. as is, MSSQL application no longer fail e2e. However parks-app can on rare occasion fail and it will always be registry 503 error.
```bash
Pod not running with condition: containers with unready status: [restify]
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: InProgress
backup phase: PartiallyFailed
[91m[1m• Failure [230.849 seconds][0m
AWS backup restore tests
[90m/Users/hiteshwaripatel/oadp-operator/tests/e2e/backup_restore_suite_test.go:16[0m
  backup and restore applications
  [90m/Users/hiteshwaripatel/oadp-operator/tests/e2e/backup_restore_suite_test.go:60[0m
    [91m[1mParks application >=4.8.0 [It][0m
    [90m/Users/hiteshwaripatel/oadp-operator/tests/e2e/backup_restore_suite_test.go:162[0m

    [91mExpected
        <[]string | len:1, cap:1>: [
            "velero container error line#1101: time=\"2021-10-12T15:03:12Z\" level=error msg=\"Error backing up item\" backup=openshift-adp/parks-e2e-0d147a74-2b6d-11ec-a348-aa665a04cd65 error=\"error executing custom action (groupResource=imagestreams.image.openshift.io, namespace=parks-app, name=restify): rpc error: code = Unknown desc = Error trying to reuse blob sha256:4b21dcdd136d133a4df0840e656af2f488c226dd384a98b89ced79064a4081b4 at destination: error pinging docker registry oadp-ts-velero-sample-1-aws-registry-route-openshift-adp.apps.ci-op-8iqn61dw-cf52a.origin-ci-int-aws.dev.rhcloud.com: invalid status code from registry 503 (Service Unavailable)\" error.file=\"/go/src/github.com/vmware-tanzu/velero/pkg/backup/item_backupper.go:331\" error.function=\"github.com/vmware-tanzu/velero/pkg/backup.(*itemBackupper).executeActions\" logSource=\"pkg/backup/backup.go:435\" name=restify\n",
        ]
    to equal
        <[]string | len:0, cap:0>: [][0m

    /Users/hiteshwaripatel/oadp-operator/tests/e2e/backup_restore_suite_test.go:89
```

File: tests/e2e/restore.go [hunk 1]
```diff
@@ -51,5 +57,5 @@ func isRestoreCompletedSuccessfully(ocClient client.Client, veleroNamespace, nam
 	if restore.Status.Phase == velero.RestorePhaseCompleted {
 		return true, nil
 	}
-	return false, nil
+	return false, fmt.Errorf("restore phase is: %s; expected: %s\nfailure reason: %s", restore.Status.Phase, velero.RestorePhaseCompleted, restore.Status.FailureReason)
```
Comment (weshayutin) [inline tests/e2e/restore.go]: nice :)

File: tests/e2e/backup_restore_suite_test.go [hunk 2]
```diff
@@ -29,6 +29,14 @@ var _ = Describe("AWS backup restore tests", func() {
 
 		err = vel.CreateOrUpdate(&vel.CustomResource.Spec)
 		Expect(err).NotTo(HaveOccurred())
+
+		log.Printf("Waiting for velero pod to be running")
+		Eventually(isVeleroPodRunning(namespace), time.Minute*3, time.Second*5).Should(BeTrue())
```
Comment (weshayutin) [inline tests/e2e/backup_restore_suite_test.go:34]: This is awesome.. is it possible to kick the time vars out to a config file in a future revision?  In the case one environment is faster/slower than others? ++   THANK YOU!!!

File: tests/e2e/backup_restore_suite_test.go [hunk 3]
```diff
@@ -29,6 +29,14 @@ var _ = Describe("AWS backup restore tests", func() {
...
+
+		if vel.CustomResource.Spec.EnableRestic == nil || *vel.CustomResource.Spec.EnableRestic {
+			log.Printf("Waiting for restic pods to be running")
+			Eventually(areResticPodsRunning(namespace), time.Minute*3, time.Second*5).Should(BeTrue())
```
Comment (weshayutin) [inline tests/e2e/backup_restore_suite_test.go:38]: same here.

**Comment by hhpatel14** [comment]
Ran E2E test with code change consecutively 15 times on 4.7 and 4.9.

**Pull request merged**